File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if(root == null) {return 0;}
        int leftHeigth = minDepth(root.left);
        int rightHeigth = minDepth(root.right);
        if(root.left == null && root.right == null) return 1;
        if(root.left == null){
            return rightHeigth + 1;
        }
        if(root.right == null){
            return leftHeigth + 1;
        }
        
        return Math.min(leftHeigth, rightHeigth) + 1;

    }
}

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        // if root is NULL return 0
        if(root == NULL ) return 0;
        // Initialize the depth of 2 subtrees
        int leftDepth = minDepth(root->left);
        int rightDepth = minDepth(root->right);
        // if root->left and root->right are NULL return 1
        if(root->left == NULL && root->right == NULL) return 1;
        // if root->left  is NULL rightDepth +1
        if(root->left == NULL ) {
            return rightDepth + 1;
        }
        // if root->right  is NULL leftDepth +1
        if(root->right == NULL ) {
            return leftDepth + 1;
        }
        return min(leftDepth, rightDepth)+1;
    }
};